test/yargs.js

```diff
@@ -2035,17 +2058,23 @@ describe('yargs dsl tests', () => {
     })
 
     it('allows a defaultDescription to be set', () => {
-      yargs('cmd')
+      const r = checkOutput(() => yargs('cmd --help').wrap(null)
         .command('cmd [heroes...]', 'a command', (yargs) => {
```

Comment:
I'm a tiny bit afraid of this change, as it seems to potentially be breaking (not a blocker, but might mean this needs to be released in a major).

Is the issue that `yargs.getOptions().defaultDescription` used to maintain state from the last command to execute, and now with this fix the freezing behavior reverts it to an initial state?

Reply:
That's exactly the point.

And the test I corrected was using the command state leaking to the parent yargs to perform its checks.

So the fix does not break any documented use case... but it could break some programs relying on that leaking command state, as our own test did.

So... major release or not?

Reply:
I say major, not a big deal, we're at 13.x already :stuck_out_tongue:

With tools like renovate bot and greenkeeper, I'd rather stay on the side of caution ... it lets us catch major bugs sooner.